src/ui/pages/NotificationDetails/components/MultiSigRequest.tsx

```diff
@@ -0,0 +1,297 @@
+import {
+  IonCard,
+  IonList,
+  IonItem,
+  IonGrid,
+  IonRow,
+  IonCol,
+  IonLabel,
+  IonIcon,
+  IonSpinner,
+} from "@ionic/react";
+import { personCircleOutline } from "ionicons/icons";
+import { useCallback, useEffect, useState } from "react";
+import { Alert as AlertDecline } from "../../../components/Alert";
+import KeriLogo from "../../../assets/images/KeriGeneric.jpg";
+import { useAppDispatch, useAppSelector } from "../../../../store/hooks";
+import {
+  getIdentifiersCache,
+  setIdentifiersCache,
+} from "../../../../store/reducers/identifiersCache";
+import { Agent } from "../../../../core/agent/agent";
+import {
+  IdentifierShortDetails,
+  MultiSigIcpRequestDetails,
+} from "../../../../core/agent/services/identifier.types";
+import { ToastMsgType } from "../../../globals/types";
+import { ScrollablePageLayout } from "../../../components/layout/ScrollablePageLayout";
+import { PageHeader } from "../../../components/PageHeader";
+import { i18n } from "../../../../i18n";
+import { PageFooter } from "../../../components/PageFooter";
+import "./MultiSigRequest.scss";
+import { CreateIdentifierResult } from "../../../../core/agent/agent.types";
+import { NotificationDetailsProps } from "../NotificationDetails.types";
+import {
+  getNotificationsCache,
+  setNotificationsCache,
+} from "../../../../store/reducers/notificationsCache";
+
+const MultiSigRequest = ({
+  pageId,
+  activeStatus,
+  notificationDetails,
+  handleBack,
+}: NotificationDetailsProps) => {
+  const dispatch = useAppDispatch();
+  const notificationsCache = useAppSelector(getNotificationsCache);
+  const [notifications, setNotifications] = useState(notificationsCache);
+  const [spinner, setSpinner] = useState(false);
+  const identifiersData = useAppSelector(getIdentifiersCache);
+  const [alertDeclineIsOpen, setAlertDeclineIsOpen] = useState(false);
+  const [multisigIcpDetails, setMultisigIcpDetails] =
+    useState<MultiSigIcpRequestDetails | null>(null);
+
+  const getDetails = async () => {
+    const details = await Agent.agent.multiSigs.getMultisigIcpDetails(
+      notificationDetails.a.d as string
+    );
+    setMultisigIcpDetails(details);
+  };
+
+  useEffect(() => {
+    getDetails();
+  });
+
+  const handleNotificationUpdate = async () => {
+    const updatedNotifications = notifications.filter(
+      (notification) => notification.id !== notificationDetails.id
+    );
+    setNotifications(updatedNotifications);
+    dispatch(setNotificationsCache(updatedNotifications));
+  };
+
+  const actionAccept = async () => {
+    document?.querySelector("ion-router-outlet")?.classList.add("blur");
+    setSpinner(true);
+    if (!(notificationDetails && multisigIcpDetails)) {
```

Comment:
If this is a scenario that should never happen, we should be throwing an error really. Otherwise it's impossible to debug what's going on

Reply:
The check for `notificationDetails` can be removed because this is a mandatory prop.

For `multisigIcpDetails`, it's not a scenario that should occur because the button shouldn't be clickable until those are in place, but still better to cover it in case something else regresses.

Something like `Cannot accept a multi-sig inception event before details are loaded from core`